### ISSUE TITLE
Fails to Restore Wallet Balance from Old Brave to New Brave, 0 BAT was recovered

### ISSUE THREAD

**Issue by BobT36** [brave/brave-browser#2694]
## Description 
Attempting to Restore Wallet (with 85 BAT) from Old Brave (v0.25.2) to New Brave (0.58.16) gives the message: "Wallet restored! 0 BAT was recovered. Your wallet key has been verified and loaded successfully."

## Steps to Reproduce

   1. Open Old Brave, Navigate to Brave Payments, Click Cog, Backup Wallet, Copy Wallet Seed. 
   2. Open New Brave, Navigate to Brave Payments, Click Cog.
   3. Select Restore, paste Seed, click Restore.
   4. Receive message: "Wallet restored! 0 BAT was recovered. Your wallet key has been verified and loaded successfully."

## Actual result:
![wallet](https://user-images.githubusercontent.com/14024619/50364469-f4e61300-0567-11e9-9cf0-5f3244c5281e.PNG)

![message](https://user-images.githubusercontent.com/14024619/50364476-fd3e4e00-0567-11e9-99c7-f2cc43a101ef.PNG)

## Expected result:
That my 85 BAT would import...

## Reproduces how often: 
Every Time.

## Brave version (brave://version info)
Brave: 0.58.16 Chromium: 71.0.3578.98 (Official Build) (64-bit)
Revision: 15234034d19b85dcd9a03b164ae89d04145d8368-refs/branch-heads/3578@{#897}
OS: Windows

### Reproducible on current release:
Brave Browser Live Release. Unsure about Dev.

### Website problems only:  
N/A

### Additional Information
Unsure if related but had issues with the Old Brave browser not sending Payments for months (after managing to send just one successfully before that).

Issue here: https://github.com/brave/browser-laptop/issues/13156

**Comment by NejcZdovc** [comment]
@brave/legacy_qa did anybody experienced this problem?

**Comment by NejcZdovc** [comment]
@BobT36 can you please go to this url https://balance.mercury.basicattentiontoken.org/v2/wallet/{paymentId}/balance (please replace `{paymentId}` with your paymentId from `ledger_state` file which you can find in your profile folder (inside `Default` folder)). Do you still see the same amount and do you see all the remaining  grants there?

**Comment by BobT36** [comment]
It shows: 

{"altcurrency":"BAT","probi":"60000000000000000000","balance":"60.0000","unconfirmed":"0.0000","rates":

And some more with all the rates in. But the balance showing 60 is correct, as in that's what shows in my Old Brave wallet. (Remember that one DOES struggle to send that balance to my Uphold, which is a separate topic). 

I used the "ledger-state.json" file from: "C:\Users\{MyUserName}\AppData\Roaming\brave", is this correct?

**Comment by NejcZdovc** [comment]
@BobT36 try in `C:\Users{MyUserName}\AppData\Roaming\BraveSoftware` as well

**Comment by BobT36** [comment]
Hmm I don't have that folder in AppData\Roaming. Just "Brave". 

I do, however have one in AppData\Local at: C:\Users\{MyUserName}\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default\ledger_state.json
(Note the underscore rather than - )

The balance file for this does show 0 balance. Is the folder for the new Browser, or something it's getting mixed up with the old one, or normal for AppData\Local?: 
{"altcurrency":"BAT","probi":"0","balance":"0.0000","unconfirmed":"0.0000","rates":{"BTC":0.00003446,"ETH":0.000889864428663654,"XRP":0.36714255273812063,"BCH":0.00081463795182147,"LTC":0.004192724175690473,"DASH":0.001624935162917904,"BTG":0.010159198113207547,"USD":0.13293224006659882,"EUR":0.11662025782026654},"parameters":{"adFree":{"currency":"BAT","fee":{"BAT":30},"choices":{"BAT":[20,30,50,85,100]},"range":{"BAT":[20,100]},"days":30}}}

**Comment by kjozwiak** [comment]
@BobT36 are you still having this issue? Did you manage to restore your into `brave-core`?

**Comment by BobT36** [comment]
Nope. Though I've just clicked "Backup" on the New Brave, and the seed is indeed the same.

Could it be that the BAT I have in my Old Brave (60 currently) is from the grants, and that they don't transfer over? Or should it work anyway.

**Comment by yodaross** [comment]
Had difficulty watching bloombergon brave

Same with you?

On Jan 23, 2019 2:34 PM, "Bob" <notifications@github.com> wrote:

Nope. Though I've just clicked "Backup" on the New Brave, and the seed is
indeed the same.

Could it be that the BAT I have in my Old Brave (60 currently) is from the
grants, and that they don't transfer over? Or should it work anyway.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/brave/brave-browser/issues/2694#issuecomment-456936459>,
or mute the thread

**Comment by kjozwiak** [comment]
> Had difficulty watching bloombergon brave

This issue is about issues restoring an old wallet from `muon` into the new version of Brave. If you're having issues watching videos on Bloomberg, please create a new [issue](https://github.com/brave/brave-browser/issues/new) 👍

**Comment by einsteinx2** [comment]
I'm also reproducing this 100% of the time. My main machine is macOS. I was able to restore my balance correctly when I upgraded to a new MacBook Pro, but when I use my restore phrase on my Windows 10 PC, two different Windows 10 VMs, or a Linux Mint 19.2 VM it always says successful but shows a balance of 0.

If I go to brave://rewards, on the right it says: "Wallet restored! 0 BAT was recovered. Your wallet key has been verified and loaded successfully."

It also shows me the verify link even though I've already verified.

If I check the restore phrases, they match, yet I only have the BAT on my main Mac...

**Comment by PrimarchAlpharius** [comment]
I also have this issue, after switching OS from Ubuntu 19.04 to Ubuntu 19.10.
Granted, the amount to be restored was very small, so I don't particularly care about it, but it's definitely something that could potentially become a big problem in the future.

**Comment by Paedow** [comment]
I have the same problem, I backed up my wallet seed, then made a fresh windows installation and after importing the wallet seed it says 0 BAT

brave version: Version 1.5.123 Chromium: 80.0.3987.163 (Official Build) (64-bit)

This balance query shows 0 for me. I also found the payment id in AppData\Local , the old brave is obviously deleted after a fresh install.
{"altcurrency":"BAT","probi":"0","cardBalance":"0","balance":"0.0000","unconfirmed":"0.0000","rates":....

How can I restore my balance?
